icu4j/main/classes/core/src/com/ibm/icu/impl/units/SingleUnitImpl.java

```diff
@@ -70,6 +68,7 @@ public String getNeutralIdentifier() {
             result.append(posPower);
             result.append('-');
         } else {
+            // TODO: IllegalArgumentException might not be appropriate here
```

Comment:
In my opinion, throwing IllegalArgumentException is not a good choice here. This method does not take any argument. I did not change the exception type for this time although.

Reply:
Thanks! In the next three months, we'll do a "technical debt sweep" to try to clean up the TODOs in units-related code.